From 930d0cb866b9bd53bbaf9e44e9176daf84f40e1e Mon Sep 17 00:00:00 2001
From: Eric Anholt <eric@anholt.net>
Date: Tue, 26 Jan 2010 09:43:10 -0800
Patch-mainline: 2.6.34
References: fate#310916
Git-commit: b397c836eff58cd9a43f7bd8b853a51b3ecc3420
Subject: [PATCH 0375/2588] drm/i915: Don't reserve compatibility fence regs
 in KMS mode.

The fence start is for compatibility with UMS X Servers before fence
management.  KMS X Servers only started doing tiling after fence
management appeared.

Signed-off-by: Eric Anholt <eric@anholt.net>
(cherry picked from commit b397c836eff58cd9a43f7bd8b853a51b3ecc3420)

Signed-off-by: Takashi Iwai <tiwai@suse.de>
---
 drivers/gpu/drm/i915/i915_gem.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 0fcbf24..03680f5 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -4831,7 +4831,8 @@ i915_gem_load(struct drm_device *dev)
 	}
 
 	/* Old X drivers will take 0-2 for front, back, depth buffers */
-	dev_priv->fence_reg_start = 3;
+	if (!drm_core_check_feature(dev, DRIVER_MODESET))
+		dev_priv->fence_reg_start = 3;
 
 	if (IS_I965G(dev) || IS_I945G(dev) || IS_I945GM(dev) || IS_G33(dev))
 		dev_priv->num_fence_regs = 16;
-- 
1.7.6

